src/evict/evict_lru.c

```diff
@@ -2153,7 +2153,7 @@ __evict_walk_prepare(WT_SESSION_IMPL *session, uint32_t *walk_flagsp)
             if (S2C(session)->evict->use_npos_in_pass)
                 /* Alternate with rand_prev so that the start of the tree is visited more often */
                 goto rand_prev;
-            else
+            else if (!S2C(session)->evict_legacy_page_visit_strategy)
```

Comment:
with` legacy page visit strategy` I would also not want npos to usage to affect traversal.

Reply:
We should have the flexibility to go back to legacy traversal even with npos usage

Reply:
Done